From ebc6e80ffbe1636859f6052a66dd669b7c1832f9 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:10:26 -0400
Subject: [usb] core: let dynamic ids override static ids

Message-id: <1410887463-20674-555-git-send-email-dzickus@redhat.com>
Patchwork-id: 95452
O-Subject: [RHEL7 PATCH 554/591] usb: core: let dynamic ids override static ids
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 31c6bf70955dda6ef92ab40624f289576cff97d3
Author: Bjorn Mork <bjorn@mork.no>
Date:   Sat Jan 11 02:04:00 2014 +0100

    usb: core: let dynamic ids override static ids

    This modifies the probing order so that any matching
    dynamic entry always will be used, even if the driver
    has a matching static entry.

    It is sometimes useful to dynamically update existing
    device entries. With the new ability to set the dynamic
    entry driver_info field, this can be used to test new
    additions to class driver exception lists or proposed
    changes to existing static per-device driver_info
    entries.

    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/driver.c b/drivers/usb/core/driver.c
index a525f3a..4aeb100 100644
--- a/drivers/usb/core/driver.c
+++ b/drivers/usb/core/driver.c
@@ -312,9 +312,9 @@ static int usb_probe_interface(struct device *dev)
   return error;
  }
 
- id = usb_match_id(intf, driver->id_table);
+ id = usb_match_dynamic_id(intf, driver);
  if (!id)
-  id = usb_match_dynamic_id(intf, driver);
+  id = usb_match_id(intf, driver->id_table);
  if (!id)
   return error;
 
-- 
1.7.1